File: kaji/sdk/kaji/integrations/registry/echo/echo.ts
/**
 * Echo integration. The simplest possible Kaji integration.
 *
 * Two pure functions, no auth, no network. Installed by `kaji add echo`.
 */
import { functionTool } from "@kaji/sdk";
import { z } from "zod";

export const say = functionTool(
  {
    description: "Return the input string unchanged.",
    parameters: z.object({ message: z.string() }),
    risk: "read",
  },
  async ({ message }) => ({ message }),
);

export const shout = functionTool(
  {
    description: "Return the input string uppercased.",
    parameters: z.object({ message: z.string() }),
    risk: "read",
  },
  async ({ message }) => ({ message: message.toUpperCase() }),
);

File: kaji/sdk/kaji/integrations/registry/echo/echo.py
"""Echo integration. The simplest possible Kaji integration.

Two pure functions, no auth, no network. Installed by `kaji add echo`.
"""

from __future__ import annotations

import kaji


@kaji.function_tool(description="Return the input string unchanged.", risk="read")
async def say(message: str) -> dict:
    return {"message": message}


@kaji.function_tool(description="Return the input string uppercased.", risk="read")
async def shout(message: str) -> dict:
    return {"message": message.upper()}
